tests/test_audit.py

```diff
@@ -174,11 +176,36 @@ async def test_get_audit_lookback_periods(
         lookback_period=lookback_period,
     )
 
+    expected_dates = generate_date_range(lookback_period)
+
+    # Mock the first date call with the expected_path
     gh_rest_mock.get_enterprise_audit_logs(
         status_code=200,
         search_phrase=expected_path,
         json=GITHUB_AUDIT,
     )
 
-    all_records = [record async for record in extractor.extract_records()]
-    assert all_records == GITHUB_EXPECTED_OUTPUT
+    # Mock additional dates if there are more than one
+    test_dates = expected_dates[:3] if len(expected_dates) > 3 else expected_dates
+    if len(test_dates) > 1:
+        for date in test_dates[1:]:
+            search_phrase = f"action:protected_branch.create created:{date}"
+            gh_rest_mock.get_enterprise_audit_logs(
+                status_code=200,
+                search_phrase=search_phrase,
+                json=GITHUB_AUDIT,
+            )
+
+    # replacing generate_date_range with test dates
+    # so that we don't iterate through all dates
+    import nodestream_github.client.githubclient as client_module
```

Comment:
why would you need this? What is the problem with hitting all dates?

Reply:
Oh I see, well, you can test the years by doing your more exhaustive testing directly on `generate_date_range` and then test that this handles `0,1,n` dates fine (try to avoid testing more than one thing at once)